0051-options-local-pragma.rst

```diff
@@ -0,0 +1,106 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+.. sectnum::
+.. contents::
+
+We propose to add an opportunity to switch GHC-options locally. For instance, this code
+::
+ f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
```

Comment:
Might be a good idea to add:

```haskell
{-# OPTIONS_GHC -Wno-name-shadowing #-}
```

at the top, so that we do not rely on what warnings are switched on by default.